pkg/util/codec/codec.go

```diff
@@ -293,17 +289,17 @@ func sizeInt(comparable1 bool) int {
 // EncodeKey appends the encoded values to byte slice b, returns the appended
 // slice. It guarantees the encoded value is in ascending order for comparison.
 // For decimal type, datum must set datum's length and frac.
-func EncodeKey(sc *stmtctx.StatementContext, b []byte, v ...types.Datum) ([]byte, error) {
-	return encode(sc, b, v, true)
+func EncodeKey(typeCtx types.Context, errctx errctx.Context, b []byte, v ...types.Datum) ([]byte, error) {
```

Comment:
How about simplifying the methods `EncodeKey` and `EncodeValue` to:

```
func EncodeKey(loc *time.Location, b[]byte, v ...types.Datum)
func EncodeValue(loc *time.Location, b []byte, v ...types.Datum)
```

And we should handle the error outside encode methods like this:

```
b, err := EcodeKey(loc, b, v);
err = ectx.HandleError(err, err, err)
```

Reply:
Done.